### PR TITLE
[JW8-6113] Implement progressive AES decryption

### PR DESCRIPTION
**Pull request by johnBartos**
### This PR will...
- Implement progressive software AES decryption
    - Generates new IV based on the previous chunk
    - WebCrypto throws an exception when progressive; therefore, this is software only
- TypeScript decryption classes

- Breaking changes:
    - Always uses software decryption if `enableSoftwareAES` is true
    - Sets `enableSoftwareAES` to true if using the progressive Fetch loader

### Why is this Pull Request needed?
So that we can progressively parse AES-128 encrypted streams

### Are there any points in the code the reviewer needs to double check?
Is the removal of the `destroy` method from the software decrypter OK? I don't believe it's necessary to set those members to null when the reference to the decrypter itself is set to null. Nulling that reference should cause its members to be GC'd

### Resolves issues:
JW8-6113

### CODE REVIEW

**Comment by johnBartos** [comment]
test this please

File: src/crypt/decrypter.ts [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+export default class Decrypter {
+  private logEnabled: boolean = true;
+  private observer: any;
+  private config: any;
+  private removePKCS7Padding: boolean;
+  private disableWebCrypto: boolean;
+  private subtle: boolean = false;
+  private softwareDecrypter: AESDecryptor | null = null;
+  private key: any | null = null;
```
Comment (zetagame) [inline src/crypt/decrypter.ts]: will key/fastAesKey ever not be a string or null?

Comment (johnBartos) [reply src/crypt/decrypter.ts]: addressed

File: tests/unit/controller/buffer-operation-queue.ts [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
 import * as chai from 'chai';
 import * as sinonChai from 'sinon-chai';
+chai.use(sinonChai);
+const expect = chai.expect;
 
 import BufferOperationQueue from '../../../src/controller/buffer-operation-queue';
 import { BufferOperation, SourceBuffers } from '../../../src/types/buffer';
 
-chai.use(sinonChai);
-const expect = chai.expect;
```
Comment (robwalch) [inline tests/unit/controller/buffer-operation-queue.ts]: It's odd to move these. All import statements should come first. Not doing so suggests import side-effects.

File: src/crypt/decrypter.ts [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+  private subtle: boolean = false;
+  private softwareDecrypter: AESDecryptor | null = null;
+  private key: ArrayBuffer | null = null;
+  private fastAesKey: FastAESKey | null = null;
+  private remainderData: Uint8Array | null = null;
+  private currentIV: ArrayBuffer | null = null;
+  private currentResult: ArrayBuffer | null = null;
+
+  constructor (observer, config, { removePKCS7Padding = true } = {}) {
+    this.logEnabled = true;
```
Comment (robwalch) [inline src/crypt/decrypter.ts]: No need to assign `logEnabled` a value in the property definition and here in the constructor.

**Pull request merged**